docs/e2e/extension-e2e-guidelines.md

```diff
@@ -314,3 +315,50 @@ Test Name: Connects to a Hardware wallet for Trezor
 ```
 
 ✅ Proposed solution: The Trezor import flow involves opening the Trezor website, then the user takes additional steps on that website to connect the device. A fake version of this website could be created for testing, and the test build could be updated to use this fake version. It's also worth investigating a phishing detection solution, such as replacing Github.com with an empty page.
+
+## Page Object Model (POM)
+
+POM, or Page Object Model, is a design pattern commonly utilized for automating test cases.
+
+The Page Object is an object-oriented class that acts as an interface for the page of the application under test. Page classes contain web elements and methods to interact with these elements. While automating test cases, we create objects of these Page Classes and interact with web elements by calling the methods of these classes.
+
+### Composition of Each Class Page
+
+Each class page is composed of:
+
+- **Selectors**: HTML elements.
+- **Action Methods**: Methods for interacting with the elements.
+- **Check Methods**: Methods that assert the status of elements.
+
+The testcase spec creates the necessary page class objects and utilizes their methods to perform actions.
+
+### Best Practices
+
+- A Page Object should represent meaningful elements of a page and not necessarily a complete page. It can also be a small component, like a navbar.
+- All selectors should be kept in the page class file. Carefully define selectors for elements in page classes, opting for robust selectors since they will be extensively used across various locations.
+- The page class should contain properties and methods, or be composed of objects that expose access.
+- Page Objects should remain independent and not invoke other Page Objects to prevent circular references, ensuring they are typically isolated from each other. For handling complex workflows that require interaction across multiple pages, "Processes" should be implemented. This approach enables the incorporation of all relevant Page Objects to support specific flows, such as sending a transaction or creating a swap. A dedicated "Processes" folder is used to organize and manage these complex workflows.
+- The test cases should only call Page Object methods or Processes; they don’t interact directly with page elements.
+- Page Object methods should include detailed logs and detailed error messages in all check methods to aid in debugging tests.
+- Assertions should be included within `check_` methods. In the test specifications, `check_` methods should be called rather than making assertions directly. The reason is to make `check_` methods reusable across different test cases, with enhanced logging defined within these `check_` methods.
```

Comment:
- "Assertions should be included within `check_` methods" — it sounds like other things can go inside of `check_` methods. If not, then perhaps it's worth rephrasing this slightly.
- "Specifications" here seems redundant to me.
- It's unclear to me how "enhanced logging" makes `check_` methods reusable across different tests. I've rephrased the last sentence a bit more smoothly, but feel free to revise.

``` suggestion
- Place assertions inside of `check_` methods, and call `check_` methods inside of tests instead of making assertions directly. Along with enhanced logging, this ensures that `check_` methods are reusable across different tests.

Reply:
Thanks for the suggested change ! 
I added example to show how "enhanced logging" makes `check_ `methods reusable across different tests. Please check.